server.js

```diff
@@ -1,26 +1,102 @@
 import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
+import dotenv from "dotenv";
+import listEndpoints from "express-list-endpoints";
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+
+dotenv.config()
+const mongoUrl = process.env.MONGO_URL || `mongodb+srv://${process.env.MONGO_USER}:${process.env.MONGO_PW}@cluster0.lmkbhok.mongodb.net/happythoughtsAPI?retryWrites=true&w=majority`;
+// "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+
 const port = process.env.PORT || 8080;
 const app = express();
 
+//an express middleware to check if mongoose is ready and throw an error if not. Note: Should always be added before the router.
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({
+      error: `Service unavailable`
+    })
+  }
+})
+
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
+//Define the model and SchemaTypes in Mangoose
+const ThougtsSchema = new mongoose.Schema({
```

Comment:
Good and clean schema! Minor thing, but Thoughts is misspelled here, would perhaps become an issue if you were to work more on this project and miss this. :)